src/components/namestable/worker-edit/worker-edit.component.tsx

```diff
@@ -0,0 +1,122 @@
+  container: {
+    height: "93%",
+  },
+});
+
+export function WorkerEditComponent({
+  name,
+  mode,
+  setOpen,
+}: WorkerEditComponentOptions): JSX.Element {
+  const classes = useStyles();
+
+  const dispatcher = useDispatch();
+  const { workerInfo, setWorkerInfo } = useWorkerInfo(name);
+  const [isWorkerNameValid, setIsWorkerNameValid] = useState(true);
+  const [isWorkerTimeValid, setIsWorkerTimeValid] = useState(true);
+  const [isContractTypeValid, setIsContractTypeValid] = useState(true);
+  const [isWorkerTypeValid, setIsWorkerTypeValid] = useState(true);
+
+  useEffect(() => {
+    if (mode === WorkerEditComponentMode.EDIT) {
+      workerInfo.previousWorkerName = name;
+    }
+  }, [mode, workerInfo, name]);
+  //#region event handlers
+  function handleWorkerNameUpdate(newWorkerName: string): void {
+    setWorkerInfo(workerInfo.withNewName(newWorkerName));
+  }
+
+  function handleWorkerTimeUpdate(newWorkerTime: number): void {
+    setWorkerInfo(workerInfo.withNewWorkerTime(newWorkerTime));
+  }
+
+  function handleWorkerContractTypeUpdate(newContractType: ContractType): void {
+    setWorkerInfo(workerInfo.withNewContractType(newContractType));
+  }
+
+  function handleWorkerWorkerTypeUpdate(newWorkerType: WorkerType): void {
+    setWorkerInfo(workerInfo.withNewWorkerType(newWorkerType));
+  }
+  //#endregion
+
+  function canSaveWorker(): boolean {
+    return [isWorkerNameValid, isWorkerTimeValid, isContractTypeValid, isWorkerTypeValid].every(
+      (item) => item
+    );
+  }
+
+  function handleClose(): void {
+    const newWorker = workerInfo.asWorkerInfoExtendedInterface();
+    mode === WorkerEditComponentMode.ADD
+      ? dispatcher(WorkerActionCreator.addNewWorker(newWorker))
+      : dispatcher(WorkerActionCreator.modifyWorker(newWorker));
+
+    setWorkerInfo(new WorkerInfo());
+    setOpen(false);
+  }
+
+  // #region view
+  return (
+    <Grid container className={classes.container} direction="column" justify="space-between">
+      <Grid container direction="column" spacing={5}>
+        <WorkerNameEditField
+          workerName={workerInfo.workerName}
+          setWorkerName={handleWorkerNameUpdate}
+          setIsFieldValid={setIsWorkerNameValid}
+          mode={mode}
+        />
+
+        <WorkerWorkerTypeSelector
+          workerType={workerInfo.workerType}
+          setActualWorkerType={handleWorkerWorkerTypeUpdate}
+          setIsFieldValid={setIsWorkerTypeValid}
+        />
+
+        <WorkerContractTypeSelector
+          setWorkerContractType={handleWorkerContractTypeUpdate}
+          workerContractType={workerInfo.contractType}
+          setIsFieldValid={setIsContractTypeValid}
+        />
+
+        <WorkerContractTypeDependentWorkTimeSelectror
+          employementTime={workerInfo.workerTime}
+          setWorkerTime={handleWorkerTimeUpdate}
+          setIsFieldValid={setIsWorkerTimeValid}
+          workerContractType={workerInfo.contractType}
+        />
```

Comment:
Powynosiłem wszystkie pola do osobnych komponentów, ponieważ przez to że każde z pół wymagało specyficznej walidacji trudno to było obsługiwać